src/util.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) Nitrokey GmbH
+// SPDX-License-Identifier: Apache-2.0 or MIT
+
+use serde::{Deserialize, Serialize};
+use subtle::{Choice, ConstantTimeEq};
+use trussed::types::Bytes;
+
+/// Custom byte array implementation for improved serialization.
+///
+/// Per default, `[u8; N]` is serialized inefficiently as a regular slice, not as a byte string.
+/// This type makes use of the improved serialization implementation of [`Bytes`][] while ensuring
+/// the constant size `N`.
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct BytesArray<const N: usize>(Bytes<N>);
```

Comment:
I would rather take the approach of wrapping a `[u8;N]` and manually implementing `Serialize`, `Deserialize` and `Deref` on them (which I think also gives us the `ConstantTimeEq` implementation. This avoids the unnecessary overhead of the `Vec` and the runtime checks.

Maybe just take [my existing implementation](https://github.com/serde-rs/bytes/pull/28/files#diff-efec21fceac87b5b012b9ccfdc36416fa83ed4b76f02bbb7bc908f986bbc1a06)?

Reply:
That was my first approach, but I’d rather not manually implement `Deserialize` here.  That just duplicates code from `serde-bytes` or `heapless` and requires more testing than just wrapping `Bytes`.